File: hammer-interaction/src/main/java/dev/vini2003/hammer/interaction/registry/common/HIEvents.java
package dev.vini2003.hammer.interaction.registry.common;

import dev.vini2003.hammer.interaction.api.common.manager.InteractionRuleManager;
import net.fabricmc.fabric.api.event.player.*;
import net.fabricmc.fabric.api.networking.v1.ServerPlayConnectionEvents;

public class HIEvents {
	public static void init() {
		AttackBlockCallback.EVENT.register(new InteractionRuleManager.AttackBlockListener());
		AttackEntityCallback.EVENT.register(new InteractionRuleManager.AttackEntityListener());
		
		UseBlockCallback.EVENT.register(new InteractionRuleManager.UseBlockListener());
		UseEntityCallback.EVENT.register(new InteractionRuleManager.UseEntityListener());
		UseItemCallback.EVENT.register(new InteractionRuleManager.UseItemListener());
		
		ServerPlayConnectionEvents.JOIN.register(new InteractionRuleManager.JoinListener());
		
		PlayerBlockBreakEvents.BEFORE.register(new InteractionRuleManager.BeforeBreakListener());
	}
}

File: hammer-interaction/src/main/kotlin/dev/vini2003/hammer/interaction/registry/common/HIEvents.kt
package dev.vini2003.hammer.interaction.registry.common

import dev.vini2003.hammer.interaction.common.manager.InteractionRuleManager
import net.fabricmc.fabric.api.event.player.*
import net.fabricmc.fabric.api.networking.v1.ServerPlayConnectionEvents

object HIEvents {
	fun init() {
		AttackBlockCallback.EVENT.register(InteractionRuleManager.AttackBlockCallbackListener)
		AttackEntityCallback.EVENT.register(InteractionRuleManager.AttackEntityCallbackListener)
		
		UseBlockCallback.EVENT.register(InteractionRuleManager.UseBlockCallbackListener)
		UseEntityCallback.EVENT.register(InteractionRuleManager.UseEntityCallbackListener)
		UseItemCallback.EVENT.register(InteractionRuleManager.UseItemCallbackListener)
		
		ServerPlayConnectionEvents.JOIN.register(InteractionRuleManager.ServerPlayConnectionEventsJoinListener)
		
		PlayerBlockBreakEvents.BEFORE.register(InteractionRuleManager.PlayerBlockBreakEventsBeforeListener)
	}
}